### PR TITLE
Replaced colons with periods in file names and zip file names

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

**Review by fightingtexasaggie — APPROVED** [review]
tested and works, however, we should probably slap the user's hand on the front end if they enter undesirable characters too

**Comment by mickel1138** [comment]
> tested and works, however, we should probably slap the user's hand on the front end if they enter undesirable characters too

Add a ticket for it.

**Comment by mickel1138** [comment]
I don't think we should do this, the API should just outright reject files with colons in it. Is this even possible in any filesystem?

File: src/MessageHandler/ZipDatasetFilesHandler.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function __invoke(ZipDatasetFiles $zipDatasetFiles)
         $datasetSubmissionId = $zipDatasetFiles->getDatasetSubmissionId();
         $datasetSubmission = $this->entityManager->getRepository(DatasetSubmission::class)->find($datasetSubmissionId);
         $filesInfo = $this->fileRepository->getFileNameAndPath($fileIds);
-        $destinationPath = $this->downloadDirectory . DIRECTORY_SEPARATOR . $datasetSubmission->getDataset()->getUdi() . '.zip';
+        $destinationPath = $this->downloadDirectory . DIRECTORY_SEPARATOR . str_replace(':', '.', $datasetSubmission->getDataset()->getUdi()) . '.zip';
```
Comment (mickel1138) [inline src/MessageHandler/ZipDatasetFilesHandler.php:99]: This can stay tho...

File: src/Controller/Api/DatasetSubmissionController.php [hunk 2]
```diff
@@ -378,7 +378,7 @@ public function addFile(DatasetSubmission $datasetSubmission, Request $request,
             $datasetSubmission->setFileset($fileset);
         }
         $newFile = new File();
-        $newFile->setFileName(trim($request->get('name')));
+        $newFile->setFileName(trim(str_replace(':', '.', $request->get('name'))));
```
Comment (mickel1138) [inline src/Controller/Api/DatasetSubmissionController.php:381]: I wouldn't do this here, or at all. The API should reject files with colons I feel. We can't change expectations.

Comment (praneethpr) [reply src/Controller/Api/DatasetSubmissionController.php:381]: Most operating systems, don't allow file creation with ":". Atleast on macOS and windows, this is just an additional check William wanted

File: src/MessageHandler/ZipDatasetFilesHandler.php (hunk 1 above)
Comment (praneethpr) [reply src/MessageHandler/ZipDatasetFilesHandler.php:99]: William wanted the zipFileName also without the colon

**Pull request merged**